test/setup.ts

```
@@ -87,7 +87,9 @@ export async function createTestSandbox(
   };
 }
 
-export async function stopTestSandbox(ctx: TestContext): Promise<void> {
+export async function stopTestSandbox(ctx?: TestContext): Promise<void> {
+  // If sandbox startup fails, ctx may be undefined.
+  if (!ctx) return;
```

What was reason for this??

---

To avoid cleanup crashing when sandbox startup fails and ctx is undefined (so afterAll doesn’t throw and hide the real error)